services/authService.ts

```diff
@@ -37,22 +51,78 @@ export class AuthService {
     }
   }
 
-  // Verify PIN
-  static async verifyPin(pin: string): Promise<boolean> {
+  // Verify PIN with brute force protection
+  static async verifyPin(pin: string): Promise<PinVerificationResult> {
     try {
```

Comment:
`try` blocks should cover as little ground as possible. Otherwise the `catch` won't actually know what happened.